helix-rest/src/test/java/org/apache/helix/rest/server/TestClusterAccessor.java

```diff
@@ -405,7 +407,68 @@ public void testEnableDisableMaintenanceModeWithCustomFields() {
+    Assert.assertEquals(stateModelNames.size(), 6);
+
+    String oneModel = stateModelNames.get(1);
+    String twoModel = stateModelNames.get(2);
+
+    String oneModelUri = "clusters/" + cluster + "/statemodeldefs/" + oneModel;
+    String oneResult = get(oneModelUri, null, Response.Status.OK.getStatusCode(), true);
+    ZNRecord oneRecord = OBJECT_MAPPER.readValue(oneResult, ZNRecord.class);
+
+    String twoModelUri = "clusters/" + cluster + "/statemodeldefs/" + twoModel;
+    String twoResult = get("clusters/" + cluster + "/statemodeldefs/" + twoModel, null, Response.Status.OK.getStatusCode(), true);
+    ZNRecord twoRecord = OBJECT_MAPPER.readValue(twoResult, ZNRecord.class);
+
+    // delete one, expect success
+    String deleteOneUri = "clusters/" + cluster + "/statemodeldefs/" + oneRecord.getId();
+    Response deleteOneRsp = target(deleteOneUri).request().delete();
+    Assert.assertEquals(deleteOneRsp.getStatus(), Response.Status.OK.getStatusCode());
+
+    Response queryRsp = target(oneModelUri).request().get();
+    Assert.assertTrue(queryRsp.getStatus() == Response.Status.NOT_FOUND.getStatusCode());
+    // delete one again, expect success
+    Response deleteOneRsp2 = target(deleteOneUri).request().delete();
+    Assert.assertTrue(deleteOneRsp2.getStatus() == Response.Status.OK.getStatusCode());
+    // create the delete one, expect success
+    String createOneUri = "clusters/" + cluster + "/statemodeldefs/" + oneModel;
+    Response createOneRsp = target(createOneUri).request().put(Entity.entity(OBJECT_MAPPER.writeValueAsString(oneRecord), MediaType.APPLICATION_JSON_TYPE));
+    Assert.assertTrue(createOneRsp.getStatus() == Response.Status.OK.getStatusCode());
+
+    // create the delete one again, expect failure
+    Response createOneRsp2 = target(createOneUri).request().put(Entity.entity(OBJECT_MAPPER.writeValueAsString(oneRecord), MediaType.APPLICATION_JSON_TYPE));
+    Assert.assertTrue(createOneRsp2.getStatus() == Response.Status.BAD_REQUEST.getStatusCode());
+
+    // set the delete one with a modification
+    ZNRecord newRecord = new ZNRecord(twoRecord, oneRecord.getId());
+    Response setOneRsp = target(createOneUri).request().post(Entity.entity(OBJECT_MAPPER.writeValueAsString(newRecord), MediaType.APPLICATION_JSON_TYPE));
+    Assert.assertTrue(setOneRsp.getStatus() == Response.Status.OK.getStatusCode());
+
+    String oneResult2 = get(oneModelUri, null, Response.Status.OK.getStatusCode(), true);
+    ZNRecord oneRecord2 = OBJECT_MAPPER.readValue(oneResult2, ZNRecord.class);
+    Assert.assertEquals(oneRecord2, newRecord);
+
+    // set the delete one with original; namely restore the original condition
+    Response setOneRsp2 = target(createOneUri).request().post(Entity.entity(OBJECT_MAPPER.writeValueAsString(oneRecord), MediaType.APPLICATION_JSON_TYPE));
+    Assert.assertTrue( setOneRsp2.getStatus() == Response.Status.OK.getStatusCode());
```

Comment:
Remove space after the open parenthesis?

Reply:
will use helix coding style formatter